examples/features/customloadbalancer/client/customroundrobin/customroundrobin.go

```diff
@@ -23,20 +23,16 @@ import (
 	"fmt"
 	"sync/atomic"
 
+	_ "google.golang.org/grpc" // to register pick_first
 	"google.golang.org/grpc/balancer"
 	"google.golang.org/grpc/balancer/endpointsharding"
 	"google.golang.org/grpc/connectivity"
 	"google.golang.org/grpc/grpclog"
 	"google.golang.org/grpc/serviceconfig"
 )
 
-var gracefulSwitchPickFirst serviceconfig.LoadBalancingConfig
-
 func init() {
 	balancer.Register(customRoundRobinBuilder{})
-	// Hardcode a pick first with no shuffling, since this is a petiole, and
-	// that is what petiole policies will interact with.
-	gracefulSwitchPickFirst, _ = endpointsharding.ParseConfig(json.RawMessage(endpointsharding.PickFirstConfig))
```

Comment:
Wasn't this better done here to avoid a potential runtime failure?  I think with the blank import of `grpc` it should be fixed.

Also this should `panic` (or `log.Fatal`) if it returns an error.

Reply:
For some reason when I do this blank import it still registers (i.e. grpc package init) after this module gets registered. Everything depends on gRPC and not vice versa so I don't know why this init was hitting first but it was...I think it has something to do with setting a global var, which runs before other inits: https://yourbasic.org/golang/package-init-function-main-execution-order/ but it should initialize to just nil so I don't know what's happening here.

Reply:
Oh it fixed. I think the issue was this depended on balancer aggregator, which depended on graceful switch but no dependency on ...grpc, so this init would run pre grpc init since it wasn't down the dependency tree, and top level main hit grpc eventually in the depth first search.